### PR TITLE
Adding Send button to EventIDs - allows testing of evetnids from CDI.

### PR DESCRIPTION
**Pull request by dpharris**
Adds a 'Send' button to all events in the CDI, next to the 'Refresh' and 'Write' buttons.

### CODE REVIEW

File: src/org/openlcb/cdi/swing/CdiPanel.java [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import org.openlcb.cdi.impl.ConfigRepresentation;
 import org.openlcb.implementations.EventTable;
 import org.openlcb.swing.EventIdTextField;
+import org.openlcb.ProducerConsumerEventReportMessage;
+import org.openlcb.*;
```
Comment (balazsracz) [inline src/org/openlcb/cdi/swing/CdiPanel.java:12]: I don't think this import is necessary.

**Comment by dpharris** [comment]
Balazs --

Not yet.  I was wondering if we need to add any more functionality?
If not, I will clean them up (I have commented out lines, etc.).

Also, these additions required a change to OpenLCB code:

https://github.com/openlcb/OpenLCB_Java/blob/master/src/org/openlcb/cdi/swing/CdiPanel.java,
and resulting openlcb.jar,
and to JMRI code:

https://github.com/JMRI/JMRI/blob/master/java/src/jmri/jmrix/openlcb/swing/ClientActions.java

How should I push these, is it necessary to link them in their respective
repositories?

Thanks
David

**Comment by balazsracz** [comment]
you should push it to this addSend branch, then I can review the code in
the PR.

**Comment by dpharris** [comment]
Ok, will do.

David

**Comment by dpharris** [comment]
Balazs --

I have committed my changes to my branch "morebutton", and I pushed it to
openlcb_java git.

I have committed my changes to JMRI, but my push to JMRI github failed -- I
do not have permission:

***@***.*** JMRI % git push --set-upstream origin
morebutton

remote: Permission to JMRI/JMRI.git denied to dpharris.

fatal: unable to access 'https://github.com/JMRI/JMRI.git/': The requested
URL returned error: 403



Here are my JMRI diffs:

***@***.*** JMRI % git diff master morebutton

*diff --git a/java/src/jmri/jmrix/openlcb/swing/ClientActions.java
b/java/src/jmri/jmrix/openlcb/swing/ClientActions.java*

*index 8810d80915..ff15c3fc1f 100644*

*--- a/java/src/jmri/jmrix/openlcb/swing/ClientActions.java*

*+++ b/java/src/jmri/jmrix/openlcb/swing/ClientActions.java*

@@ -68,6 +68,16 @@ public class ClientActions {

                 desc = null;

             }



+            /// dph

+            public void makeSensor(String ev, String mdesc) {

+                jmri.Sensor sensor =
jmri.InstanceManager.sensorManagerInstance()

+                    .provideSensor(memo.getSystemPrefix() + "S" + ev);

+                if (mdesc.length() > 0) {

+                        sensor.setUserName(mdesc);

+                }

+                log.info("make sensor MS{} [{}]", ev, mdesc);

+            }

+

             @Override

             public void handleGroupPaneEnd(JPanel pane) {

                 if (gpane != null && evt1 != null && evt2 != null && desc
!= null) {

*diff --git a/lib/openlcb.jar b/lib/openlcb.jar*

*index ef965ba75c..8681bd4705 100644*

Binary files a/lib/openlcb.jar and b/lib/openlcb.jar differ


David

**Comment by balazsracz** [comment]
This PR is replaced by #194.

**Pull request closed without merging**